File: app/src/main/java/com/example/healthcareapp/viewmodels/SignInViewModel.java
package com.example.healthcareapp.viewmodels;

import android.database.Observable;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.databinding.ObservableBoolean;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.healthcareapp.data.repositories.AuthRepository;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;

public class SignInViewModel extends ViewModel {
    public MutableLiveData<String> email = new MutableLiveData<>("");
    public MutableLiveData<String> password = new MutableLiveData<>("");
    public MutableLiveData<Boolean> isLoading = new MutableLiveData<Boolean>(false);
    public MutableLiveData<Boolean> isAuthenticated = new MutableLiveData<Boolean>(false);
    public void login() {
        AuthRepository.getInstance().signInWithEmail(email.getValue(), password.getValue(), new OnCompleteListener<AuthResult>() {
            @Override
            public void onComplete(@NonNull Task<AuthResult> task) {
                if (task.isSuccessful()) {
                    isAuthenticated.setValue(true);
                    Log.d("AUTHENTICATION","SUCCESS");
                } else {
                    isAuthenticated.setValue(false);
                    Log.d("AUTHENTICATION","FAILED");
                }
            }
        });
    }
}

File: app/src/main/java/com/example/healthcareapp/viewmodels/SignInViewModel.kt
package com.example.healthcareapp.viewmodels

import android.database.Observable
import android.util.Log
import androidx.databinding.ObservableField
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.healthcareapp.data.repositories.AuthRepository
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import kotlinx.coroutines.runBlocking

class SignInViewModel : ViewModel() {
    val email = MutableLiveData("")
    val password = MutableLiveData("")
    val error = ObservableField<String>("")
    val isLoading = MutableLiveData(false)
    val isAuthenticated = MutableLiveData(false)

    fun login() {
        AuthRepository.getInstance().signInWithEmail(email.value, password.value,
            OnCompleteListener<AuthResult> { task ->
                if (task.isSuccessful) {
                    val id = FirebaseAuth.getInstance().currentUser?.uid
                    if (id != null) {
                        runBlocking{
                            AuthViewModel.getUserFromDB(id)
                        }
                    }
                    isAuthenticated.value = true
                    Log.d("AUTHENTICATION", "SUCCESS")
                } else {
                    isAuthenticated.value = false
                    Log.d("AUTHENTICATION", "FAILED")
                }
            })
    }
}
